control/control.go

```diff
@@ -35,51 +32,42 @@ func NewHTTPServer(cfg *Config) (*HTTPServer, error) {
 	}
 
 	mux := http.NewServeMux()
-	addr := net.UnixAddr{
-		Name: cfg.SocketPath,
-		Net: SocketType,
-	}
 
 	return &HTTPServer{
+		Addr: cfg.SocketPath,
 		mux: mux,
-		addr: addr,
-		listening: false,
 	}, nil
 }
 
-var listener net.Listener
-
 // Serve starts serving the control server
-func (s *HTTPServer) Serve() {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	// ref https://github.com/joyent/containerpilot/pull/165
-	if listener != nil {
-		return
-	}
+func (s *HTTPServer) Start(app interface{}) {
```

Comment:
That seems sensible, and you're using `interface{}` because of the circular import I assume. You can define an interface for `App` in this package and that should work. This way we also know exactly what methods of app handlers in this package will need to call too.